tests/osgi-tests/src/test/java/org/wso2/carbon/osgi/config/PropertyResolveConfigProviderOSGITest.java

```diff
@@ -0,0 +1,123 @@
+import org.wso2.carbon.container.CarbonContainerFactory;
+import org.wso2.carbon.container.options.CopyFileOption;
+import org.wso2.carbon.kernel.config.model.CarbonConfiguration;
+
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import javax.inject.Inject;
+
+import static org.ops4j.pax.exam.CoreOptions.systemProperty;
+
+/**
+ * This class test basic functionality of Configuration provider service with property replacing when variable used.
+ * i.e. sys, env, sec.
+ *
+ * @since 5.2.0
+ */
+@Listeners(PaxExam.class)
+@ExamReactorStrategy(PerClass.class)
+@ExamFactory(CarbonContainerFactory.class)
+public class PropertyResolveConfigProviderOSGITest {
+
+    private static final String DEPLOYMENT_FILENAME = "deployment.yaml";
+    private static final String SECRET_PROPERTIES_FILENAME = "secrets.properties";
+
+    @Inject
+    private ConfigProvider configProvider;
+
+    @Configuration
+    public Option[] createConfiguration() {
+        setEnvProperty("server.name", "Test WSO2 Carbon Kernel");
+        return new Option[]{ copyCarbonYAMLOption(), systemProperty("server.id").value("test-carbon-kernel"),
+                copySecretPropertiesOption() };
+    }
+
+    @Test
+    public void testPropertyResolve() throws ConfigurationException {
+        CarbonConfiguration carbonConfiguration = configProvider.getConfigurationObject(
+                CarbonConfiguration.class);
+        String id = carbonConfiguration.getId();
+        String name = carbonConfiguration.getName();
+        String tenant = carbonConfiguration.getTenant();
+
+        Assert.assertEquals(id, "test-carbon-kernel", "id should be test-carbon-kernel");
+        Assert.assertEquals(name, "Test WSO2 Carbon Kernel",
+                "name should be Test WSO2 Carbon Kernel");
+        Assert.assertEquals(tenant, "default", "tenant should be default");
+
+    }
+
+    /**
+     * Replace the existing deployment.yaml file with the file found at property-resolve directory.
+     */
+    private CopyFileOption copyCarbonYAMLOption() {
+
+        Path carbonYmlFilePath;
+
+        String basedir = System.getProperty("basedir");
+        if (basedir == null) {
+            basedir = Paths.get(".").toString();
+        }
+        carbonYmlFilePath = Paths.get(basedir, "src", "test", "resources", "property-resolve",
+                DEPLOYMENT_FILENAME);
+        return new CopyFileOption(carbonYmlFilePath, Paths.get("conf", DEPLOYMENT_FILENAME));
+    }
+
+    /**
+     * Replace the existing secrets.properties file with the file found in property-resolve directory.
+     *
+     * @return CopyFileOption object.
+     */
+    private CopyFileOption copySecretPropertiesOption() {
+        Path carbonYmlFilePath;
+
+        String basedir = System.getProperty("basedir");
+        if (basedir == null) {
+            basedir = Paths.get(".").toString();
+        }
+        carbonYmlFilePath = Paths.get(basedir, "src", "test", "resources", "property-resolve",
```

Comment:
That is because these files in this directory used for to check property resolve functionality in config component.